File: Qatime_Student/TeacherPublicClassCollectionViewCell.m
//
//  TeacherPublicClassCollectionViewCell.m
//  Qatime_Student
//
//  Created by Shin on 2016/11/9.
//  Copyright © 2016年 WWTD. All rights reserved.
//

#import "TeacherPublicClassCollectionViewCell.h"

@implementation TeacherPublicClassCollectionViewCell

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        
        
        
        /* 白色背景*/
        [self.contentView setBackgroundColor:[UIColor whiteColor]];
        
        /* 课程图布局*/
        
        _classImage = [[UIImageView alloc]init];
        //        [_classImage setImage:[UIImage imageNamed:@"school"]];
        [self.contentView addSubview:_classImage];
        _classImage.sd_layout.topSpaceToView(self.contentView,0).leftSpaceToView(self.contentView,0).rightSpaceToView(self.contentView,0).heightRatioToView(self.contentView,0.7);
        
        
        
        /* 教师姓名label*/
        /**** 预留label的宽度和教师姓名model源 ***/
        _className = [[UILabel alloc]init];
        [self.contentView addSubview:_className];
        _className.textColor = [UIColor blackColor];
        _className.sd_layout.leftSpaceToView(self.contentView,0).topSpaceToView(_classImage,5).autoHeightRatio(0);
        [_className setSingleLineAutoResizeWithMaxWidth:300];
        
        [_className setText:@"课程名称"];
        _className.textAlignment = NSTextAlignmentLeft;
        
        /* 年级 label*/
        _grade = [[UILabel alloc]init];
        [self.contentView addSubview:_grade];
        [_grade setText:@"年级"];
        _grade.textColor =[UIColor grayColor];
        _grade.sd_layout.leftEqualToView(_className).topSpaceToView(_className,0).autoHeightRatio(0);
        [_grade setSingleLineAutoResizeWithMaxWidth:80];
        _grade.textAlignment = NSTextAlignmentRight;
        [_grade setFont:[UIFont systemFontOfSize:12]];
        
        /* 科目 label*/
        /* 科目名称  预留名称的接口model*/
        _subjectName = [[UILabel alloc]init];
        [self.contentView addSubview:_subjectName];
        [_subjectName setText:@"科目名称"];
        _subjectName.textColor =[UIColor grayColor];
        _subjectName.sd_layout.leftSpaceToView(_grade,0).topSpaceToView(_className,0).autoHeightRatio(0);
        [_subjectName setSingleLineAutoResizeWithMaxWidth:80];
        _subjectName.textAlignment = NSTextAlignmentRight;
        [_subjectName setFont:[UIFont systemFontOfSize:12]];
        
       
        
        /* 价格label*/
        _priceLabel = [[UILabel alloc]init];
        [self.contentView addSubview:_priceLabel];
        [_priceLabel setText:@"¥0.00"];
        _priceLabel.textColor =[UIColor redColor];
        _priceLabel.sd_layout.rightSpaceToView(self.contentView,0).topEqualToView(_grade).autoHeightRatio(0);
        [_priceLabel setSingleLineAutoResizeWithMaxWidth:80];
        _priceLabel.textAlignment = NSTextAlignmentRight;
        [_priceLabel setFont:[UIFont systemFontOfSize:12]];

        
        
    }
    return self;
}

-(void)setModel:(TutoriumListInfo *)model{
    
    _model = model;
    _className.text = model.name;
    [_classImage sd_setImageWithURL:[NSURL URLWithString:model.publicize] placeholderImage:[UIImage imageNamed:@"school"]];
    _grade.text = model.grade;
    _subjectName.text = model.subject;
    _priceLabel.text = [NSString stringWithFormat:@"¥%@",model.price];
    
    
}


@end

File: Qatime_Student/TeacherPublicClassCollectionViewCell.swift
//
//  TeacherPublicClassCollectionViewCell.swift
//  Qatime_Student
//
//  Created by Shin on 2017/3/2.
//  Copyright © 2017年 WWTD. All rights reserved.
//

import UIKit

class TeacherPublicClassCollectionViewCell: UICollectionViewCell {
   
    //属性定义
    
    //课程名
    public var className:UILabel = UILabel()
    //背景图
    public var classImage = UIImageView()
    //年级
    public var grade = UILabel()
    //科目
    public var subjectName = UILabel ()
    //价格
    var priceLabel = UILabel()
    //model
    var model : TutoriumListInfo? {
      
        didSet{
        
            className.text = model?.name
            let url  = NSURL(string: (model?.publicize)!)
            classImage.sd_setImage(with: url as URL!)
            grade.text = model?.grade
            subjectName.text = model?.subject
            
            let price = "¥ +\(model?.price)"
            
            priceLabel.text = price
            
        }
    
    }
    
    //重写初始化方法
    override init(frame: CGRect) {
        super.init(frame: frame)
        //写各种属性方法吧
        self.backgroundColor = UIColor.white
        
        //课程图的布局
        
        contentView.addSubview(classImage)
        
        _ = classImage.sd_layout()
            .topSpaceToView(contentView,0)?
            .leftSpaceToView(contentView,0)?
            .rightSpaceToView(contentView,0)?
            .heightRatioToView(contentView,0.7)
        
        
        //教师姓名
        
        contentView.addSubview(className)
        className.textColor = UIColor.black
        className.text = "课程名称"
        className.textAlignment = .left

        _ = className.sd_layout()
            .leftSpaceToView(contentView,0)?
            .rightSpaceToView(contentView,0)?
            .topSpaceToView(classImage,5)?
            .heightIs(30)
        
        //年级
        
        contentView.addSubview(grade)
        grade.text = "年级"
        grade.textColor = UIColor.gray
        grade.textAlignment = .right
        grade.font = UIFont.systemFont(ofSize: 12)
        
        _ = grade.sd_layout()
            .leftEqualToView(className)?
            .topSpaceToView(className,0)?
            .autoHeightRatio(0)
        grade.setSingleLineAutoResizeWithMaxWidth(80)
        
        
        //科目
        
        contentView.addSubview(subjectName)
        subjectName.text = "科目名称"
        subjectName.textColor = UIColor.gray
        subjectName.textAlignment = .right
        subjectName.font = UIFont.systemFont(ofSize: 12)
        
        _=subjectName.sd_layout()
        .leftSpaceToView(grade,0)?
            .topSpaceToView(className,0)?
            .autoHeightRatio(0)
        subjectName .setSingleLineAutoResizeWithMaxWidth(80)
        
        
        //价格
        
        contentView.addSubview(priceLabel)
        priceLabel.text = "¥0.00"
        priceLabel.textColor = UIColor.red
        priceLabel.textAlignment = .right
        priceLabel.font = UIFont.systemFont(ofSize: 12)
        
        _ = priceLabel.sd_layout()
            .rightSpaceToView(contentView,0)?
            .topEqualToView(grade)?
            .autoHeightRatio(0)
        
        priceLabel.isHidden = true
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    
    
    
}
